csrc/ir/builder.cpp

```diff
@@ -240,6 +240,12 @@ Val* IrBuilder::getAttrExpr(Val* struct_, std::string attr) {
   return out;
 }
 
+Val* IrBuilder::reverseArrayExpr(Val* array) {
+  auto out = newScalar(array->dtype());
```

Comment:
Can you check if my understanding is correct? when we run the added test with 
```
  auto input = IrBuilder::create<Val>(
      DataType(ArrayOf{std::make_shared<DataType>(DataType::Int), 5}));
  auto output = IrBuilder::reverseArrayExpr(input);
```
The `array->dtype()` in this function becomes `ArrayOf of int`, then `newScalar` will create a val which is  `ArrayOf of int`. So the `scalar` in   `newScalar` is not the `scalar` in `scalar-vector-2D tensor` because here it is actually an   `ArrayOf of int`.

Reply:
You are right. The name `newScalar` was due to historical reason. It's no longer a good name today.

Reply:
Added an issue for removing `newScalar`: https://github.com/NVIDIA/Fuser/issues/754